map_model/src/objects/traffic_signals.rs

```diff
@@ -71,6 +79,27 @@ impl ControlTrafficSignal {
         brute_force(map, id)
     }
 
+    pub fn get_min_crossing_time(&self, idx: usize) -> Duration {
+        let mut max_distance = Distance::meters(0.0);
+        for movement in &self.stages[idx].protected_movements {
+            if movement.crosswalk {
+                let distance = self.movements.get(&movement).unwrap().geom.length();
+                if max_distance == Distance::meters(0.0) {
```

Comment:
I'll add a `max` fn to `Distance` and simplify this later

Reply:
Done

Reply:
Implemented